[ATSPI] Implement `AccessibilityUIElement::sortDirection()`

obyknovenius opened the pull request:
#### 8d8921074e34a5797c0cdb3c86f16899b7f1d06a
<pre>
[ATSPI] Implement `AccessibilityUIElement::sortDirection()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=250673">https://bugs.webkit.org/show_bug.cgi?id=250673</a>

Reviewed by Carlos Garcia Campos.

It is used by `accessibility/custom-elements/table.html`.

* LayoutTests/accessibility/custom-elements/table.html:
* LayoutTests/platform/gtk/accessibility/custom-elements/table-expected.txt:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.cpp:
(WTR::AccessibilityUIElement::sortDirection const): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::AccessibilityUIElement::sortDirection const):
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityUIElementWin.cpp:
(WTR::AccessibilityUIElement::sortDirection const):

Canonical link: <a href="https://commits.webkit.org/258975@main">https://commits.webkit.org/258975@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7a82cec2d6481498f4d277b713e262a033b630da

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103527 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12645 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36486 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112763 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172972 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107481 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13674 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3546 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95782 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/111938 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109300 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10532 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38264 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92353 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25192 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79905 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6028 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26594 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6208 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3121 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12188 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46104 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6166 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7962 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
```diff
@@ -456,6 +456,13 @@ JSRetainPtr<JSStringRef> AccessibilityUIElement::currentStateValue() const
+{
+    m_element->updateBackingStore();
+    auto value = m_element->attributes().get("sort"_s);
+    return OpaqueJSString::tryCreate(value).leakRef();
```
I think it's better to check the value here and convert it to what tests expect (`AXAscendingSortDirection`, `AXDescendingSortDirection` or `AXUnknownSortDirection`). This way we don't have to change every test and results can be shared.

obyknovenius committed:
[ATSPI] Implement `AccessibilityUIElement::sortDirection()`
https://bugs.webkit.org/show_bug.cgi?id=250673

Reviewed by Carlos Garcia Campos.

It is used by `accessibility/custom-elements/table.html`.

* LayoutTests/accessibility/custom-elements/table.html:
* LayoutTests/platform/gtk/accessibility/custom-elements/table-expected.txt:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.cpp:
(WTR::AccessibilityUIElement::sortDirection const): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::AccessibilityUIElement::sortDirection const):
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityUIElementWin.cpp:
(WTR::AccessibilityUIElement::sortDirection const):

Canonical link: https://commits.webkit.org/258975@main
